Mathlib/Tactic/CategoryTheory/BicategoryCoherence.lean

```diff
@@ -99,21 +99,21 @@ def exception' (msg : MessageData) : TacticM Unit := do
     -- There might not be any goals
     throwError msg
 
-set_option quotPrecheck false in
+/-- Sending 2-morphisms in `FreeBicategory B` to those in `B`. -/
+def mkLiftMap₂LiftExprAux {a b : FreeBicategory B} {f g : a ⟶ b} (η : f ⟶ g) :=
+  (FreeBicategory.lift (Prefunctor.id _)).map₂ η
+
 /-- Auxiliary definition for `bicategorical_coherence`. -/
--- We could construct this expression directly without using `elabTerm`,
--- but it would require preparing many implicit arguments by hand.
-def mkLiftMap₂LiftExpr (e : Expr) : TermElabM Expr := do
-  Term.elabTerm
-    (← ``((FreeBicategory.lift (Prefunctor.id _)).map₂ (LiftHom₂.lift $(← Term.exprToSyntax e))))
-    none
+def mkLiftMap₂LiftExpr (e : Expr) : MetaM Expr := do
+  let f ← mkAppOptM ``LiftHom₂.lift #[none, none, none, none, none, none, none, none, e, none]
+  mkAppM ``mkLiftMap₂LiftExprAux #[f]
```

Comment:
I would strongly suggest using Qq to avoid the kind of errors you have spotted and make sure they don't reappear

Reply:
Pushing in a different direction, I generally advise to avoid using `Qq` unless it is for functionality that is really missing elsewhere. 🤷‍♀️

Reply:
Not using Qq already caused bugs here though

Reply:
From another perspective, the bug should have been caught by tests. This PR improves the tests as well.

Reply:
One argument for `Qq` here is that ```` ``LiftHom₂.lift #[none, none, none, none, none, none, none, none, e, none] ``` is super fragile, and fails only in the test rather than in the elaborator if the argument order changes.

Reply:
I tried to use `Qq` this part, but I was not completely successful https://github.com/leanprover-community/mathlib4/pull/15335/commits/d886e88ba98dc4b05cc46e426f8dd7bfd1b277f0.

Instead, I would like to suggest using `MkAppM` here to avoid the `none` sequence, which I believe reasonable solutions for this situation.

Reply:
It seems that the current `Qq` doesn't work well with matching universe variables needed here. See [Zulip](https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/.2315335.20meta.20code.20for.20monoidal.20categories).

Actually, this part is independent of the other fixes, so I would like to separate it into a future PR to make it easier to review.

Reply:
Feel free to make it a separate PR yes

Reply:
Will you split the PR?

Reply:
I have made #16570.

Reply:
Are you reverting stuff from this PR accordingly?

Reply:
Yes, I separated the codes related to `Qq` discussion into #16570.